### DISCUSSION TITLE
InvenTree Mobile App - Uploading Invoice w/ Purchase Order & Deducting Stock Items

### DISCUSSION THREAD

**Discussion by palagap** [inventree/InvenTree#3121] [Q&A] [answered]
Hi there,

I've purchased the InvenTree mobile application (for iOS), and am experimenting with a couple of processes. 

First, I'm trying to use the app to complete purchase orders. I'm following the instructions listed here:

https://inventree.readthedocs.io/en/latest/app/po/

I'd like to do the following: when items from a PO are received (physically), receive those items into stock via the app and upload the corresponding invoice as an attachment. I believe the former is possible already by long-pressing on the line item in the PO interface, but I'm not sure about the latter task. Is that something that's possible? 

Another task I'd like to do is use the app to scan the barcode of a part, and then deduct a certain number of that part from the inventory (as they will be put to use). Is this feature available in the app?

Thanks.

**Comment by SchrodingersGat** [comment, marked as answer]
Hi @palagap thanks for the questions. Firstly there's a separate repo for the app - https://github.com/inventree/inventree-app - any future discussions would best be directed there.

> I'd like to do the following: when items from a PO are received (physically), receive those items into stock via the app and upload the corresponding invoice as an attachment

There is an outstanding issue to implement attachment support for purchase orders - https://github.com/inventree/inventree-app/issues/142

> Another task I'd like to do is use the app to scan the barcode of a part, and then deduct a certain number of that part from the inventory (as they will be put to use). Is this feature available in the app?

This is supported. Scan the part with the barcode scanner in the app, it will take you to the detail page where you can select the "remove stock" action to select how many items you wish to remove.

**Reply by palagap** [reply]
Great, thanks!

**Reply by palagap** [reply]
As a quick follow up, is it possible to allocate the removed stock of an item to a build from the app? E.g. rather than just deducting the stock from the overall inventory, deduct it and assign it to a build.

**Reply by SchrodingersGat** [reply]
No, the app currently does not support allocation of stock to build orders

**Reply by palagap** [reply]
Got it, thanks for the clarification.
